packages/ejson_annotation/lib/ejson_annotation.dart

```diff
@@ -0,0 +1,20 @@
+// Copyright 2023 MongoDB, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/// Annotation to mark a class for extended json (ejson) serialization
+const ejson = EJson();
+
+/// Annotation to mark a class for extended json (ejson) serialization
+class EJson<T> {
+  final EJsonEncoder<T>? encoder;
+  final EJsonDecoder<T>? decoder;
+  const EJson({this.encoder, this.decoder});
+}
+
+typedef EJsonDecoder<T> = T Function(EJsonValue ejson);
+
+typedef EJsonEncoder<T> = EJsonValue Function(T object);
+
+// while we wait for
+// typedef EJsonValue = Null | String | bool | int | double | List<EJsonValue> | Map<String, EJsonValue>;
+typedef EJsonValue = Object?;
```

Comment:
Should we provide convenience serialize/deserialize methods to EJsonValue? It may not be obvious that it is intended to be used as an intermediate product that is fed to the regular json serializer.

Reply:
We could. Something like:
```dart
toEJsonString(Object? value) => jsonEncode(toEjson(value));
fromEjsonString<T>(String ejson) => fromEJson(jsonDecode(ejson)); 
```

Reply:
Fixed